Packs/Armorblox/Integrations/Armorblox/Armorblox.py

```diff
@@ -124,49 +135,80 @@ def get_remediation_action(client, incident_id):
-    start_time: Any
-    # pageToken fetched from demisto lastRun
-    pageToken = int()
+    current_time = datetime.utcnow().replace(second=0)
+    last_fetch_time = last_run.get("last_fetch_time", None)
     incidents = []
-    if 'start_time' not in last_run.keys():
-        pageToken = -1
-        response, next_page_token = client.get_incidents(pageSize=1, pageToken=pageToken, first_fetch=FIRST_FETCH)
-        if response:
-            response = response[0]
-            start_time = response.get('date')
-            start_time = dateparser.parse(start_time)
-            message_ids = get_incident_message_ids(client, response.get('id'))
-            response['message_ids'] = message_ids
-            curr_incident = {'rawJSON': json.dumps(response), 'details': json.dumps(response)}
-            incidents.append(curr_incident)
-
-    if last_run and 'pageToken' in last_run.keys():
-        pageToken = last_run.get('pageToken')
-
-    if last_run and 'start_time' in last_run.keys():
-        start_time = dateparser.parse(last_run.get('start_time'))
-
-    start_time = start_time.timestamp()
-    incidents_data, pageToken = get_incidents_list(client, pageToken=pageToken, first_fetch=FIRST_FETCH)
-    last_time = start_time
+    incidents_data = []
+    mapping = {
+        "lastDay": 1,
+        "last3Days": 3
+    }
+    if last_fetch_time is None:
+        last_fetch_time = (current_time - timedelta(days=mapping[FIRST_FETCH])).strftime(
+            ARMORBLOX_INCIDENT_API_TIME_FORMAT)
+
+    current_time = current_time.strftime(ARMORBLOX_INCIDENT_API_TIME_FORMAT)
+    next_page_token = None
+    while True:
+        response, next_page_token = get_incidents_list(client, pageToken=next_page_token,
+                                                       from_date=last_fetch_time,
+                                                       to_date=current_time,
+                                                       first_fetch=FIRST_FETCH)
+        incidents_data.extend(response)
+        if not next_page_token:
+            break
+    last_fetch_time = current_time
 
     for incident in incidents_data:
-        dt = incident.get('date')
-        parsed_date = dateparser.parse(dt)
-        assert parsed_date is not None, f'failed parsing {dt}'
-        dt = int(parsed_date.timestamp())
-        # Update last run and add incident if the incident is newer than last fetch
-        if dt > int(start_time):
-            curr_incident = {'rawJSON': json.dumps(incident), 'details': json.dumps(incident)}
-            last_time = dt
-            incidents.append(curr_incident)
+        curr_incident = {'rawJSON': json.dumps(incident), 'details': json.dumps(incident)}
+        incidents.append(curr_incident)
     # Save the next_run as a dict with the start_time key to be stored
-    demisto.setLastRun({'start_time': str(last_time), 'pageToken': pageToken})
+    demisto.setLastRun({'last_fetch_time': last_fetch_time})
     return incidents
```

Comment:
The `fetch_incidents` code looks way better! the reason for these changes is the client method that allowed to simplify this method?
I don't see you consider the `max_fetch` parameter, return only `max_fetch` results to load balance the system.